src/adrs/0129349218/README.md

```diff
@@ -0,0 +1,506 @@
+ * highlight an invalid field in a UI or a backend service to trigger specific
+ * compensation logic.
+ */
+type Subject = string;
+
+/**
+ * ID identifies the error. The combination of info + id is unique for each
+ * distinct error event. Consumers MAY assume that error events with identical
+ * source and id are duplicates. This ID can be provided to support teams as a
+ * reference to help identify and track specific error occurrences, enabling
+ * faster troubleshooting and resolution of issues.
+ */
+type Id = string
+
+/**
+ * Time timestamp of when the occurrence happened.
+ */
+type time = Timestamp;
+
+/**
+ * Domain identifies the context in which an error happened.
+ */
+type ErrorInfo = {
+  /**
+   * The reason of the error. This is a constant value that identifies the
+   * proximate cause of the error. Error reasons are unique within a particular
+   * domain of errors. This should be at most 63 characters and match a regular
+   * expression of `[A-Z][A-Z0-9_]+[A-Z0-9]`, which represents UPPER_SNAKE_CASE.
+   */
+  reason: string;
+
+  /**
+   * The logical grouping to which the "reason" belongs. The error domain is
+   * typically the registered service name of the tool or product that generates
+   * the error. Example: "pubsub.googleapis.com". If the error is generated by
+   * some common infrastructure, the error domain must be a globally unique value
+   * that identifies the infrastructure. For Google API infrastructure, the error
+   * domain is "googleapis.com".
+   */
+  domain: string;
+
+  /**
+   * Additional structured details about this error. Values in this field MUST
+   * be carefully sanitized to avoid leaking sensitive information. See the
+   * "Secure Metadata Handling" section for detailed guidance.
+   */
+  metadata: Record<string, string>;
+};
+
+type HelpLink = {
+  /**
+   * Describes what the link offers.
+   */
+  description: string;
+
+  /**
+   * The URL of the link.
+   */
+  url: string;
+};
+
+type Help = {
+  /**
+   * URL(s) pointing to additional information on handling the current error.
+   */
+  links: Array<HelpLink>;
+};
+
+/**
+ * Contains the stack trace of the error and other debug information.
+ *
+ * This field is intended for internal logging and debugging only. It MUST be
+ * stripped from any error payload sent to an external client or end-user.
+ * Exposure of this information can lead to severe security vulnerabilities.
+ * In practice, you should only send the debug information to internal logging
+ * systems and remove it when sending the error to the client.
+ */
+type DebugInfo = {
+  /**
+   * The stack trace entries indicating where the error occurred.
+   */
+  stack_entries: string[];
+
+  /**
+   * Additional debugging information provided by the server. While this entire
+   * `DebugInfo` object is intended for internal use, care should still be
+   * taken to avoid unnecessarily logging highly sensitive data.
+   */
+  metadata: Record<string, string>;
```

Comment:
```suggestion
  metadata: Record<string, any>;
```

`any` or `unknown` if this is just for documentation purposes. An object of `{"1": "1", "2": "2"}` won't satisfy real usage. Realistically, this would likely be a generic, right? I'm assuming that in a well-typed library implementation for this you'd be able to provide a scrubbing callback for well-defined/structured data as mentioned in the `Secure Metadata Handling` section.

Reply:
Intentionally, this is made only strings, otherwise, all other infrastructures must deal with complex data.

In the worst case, I am OK with opening up the door to be compatible with OTEL attributes.

About generic, definitely you can add generics in your particular SDK if you wish, the spec doesn't care about that part.

This is where you want factory functions close to your domain to guarantee integrity. If I ever provide a JS SDK, it must be taken into consideration

Something like,

```ts
function makeGenericNotFoundError(metadata: {resource_id: string, resource_type: string}) {
  return new NotFoundError({metadata: metadata)
}
```